build.sbt

```diff
@@ -152,17 +155,45 @@ lazy val connect = (project in file("graphframes-connect"))
     assembly / assemblyShadeRules := Seq(
       ShadeRule.rename("com.google.protobuf.**" -> protobufShadingPattern).inAll),
     assembly / assemblyMergeStrategy := {
-      case PathList("google", "protobuf", xs @ _*) => MergeStrategy.discard
       case PathList("META-INF", xs @ _*) => MergeStrategy.discard
       case x if x.endsWith("module-info.class") => MergeStrategy.discard
-      case x => MergeStrategy.first
+      case _ => MergeStrategy.first
     },
-    assembly / assemblyExcludedJars := (Compile / fullClasspath).value.filter { className =>
-      className.data
-        .getName()
-        .contains("scala-library-") || className.data
-        .getName()
-        .contains("slf4j-api-")
+    // I have zero ideas how to apply shading and exclude everything except connect classes in the right way.
+    // This looks terrible, but it works:
+    assembly / assemblyExcludedJars := (Compile / fullClasspath).value.filterNot { className =>
```

Comment:
I'm leaving only classes, related to the connect plugin itself. I do not know how to do it in a more elegant way but at least it works.